lib/src/runner/vm/platform.dart

```diff
@@ -52,7 +53,38 @@ class VMPlatform extends PlatformPlugin {
   /// This isolate connects an [IsolateChannel] to [message] and sends the
   /// serialized tests over that channel.
   Future<Isolate> _spawnIsolate(String path, SendPort message) async {
-    if (_config.pubServeUrl == null) {
+    if (_config.suiteDefaults.precompiledPath != null) {
```

Comment:
[nit] this method is hard for me to follow, it looks like there are 3 top level cases. Could you see how it works to split this out?

```dart
 Future<Isolate> _spawnIsolate(String path, SendPort message) {
  if (_config.suiteDefaults.precompiledPath != null) {
    return _someNiceName(path, message);
  } else if (_config.pubServeUrl != null) {
    return _spawnIsolateWithPubUri(path, message);
  } else {
    return _someOtherNiceName(path, message);
  }
}